### ISSUE TITLE
no-useless-predicate: false-positive for 'typeof emptyObject === "object"'

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#535]
```ts
declare var v: {};

typeof v === 'object'; // reported as always true
```

The above shouldn't give an error as an empty object type can contain primitive values as well.
There's already a condition for that, but it seems that handles a different case.
